### PR TITLE
fix(rome_cli): correctly compute configuration

### LINKED ISSUE #3175: 🐛 `rome format` ignores command line options
### Environment information

```bash
Version: `main`
```

### What happened?

Run `rome` with `quote-style="single"` 

```bash
cargo run --bin rome --release format ~/Desktop/jquery.min.js --quote-style=single
```

Rome formats the string literals in `jquery.min.js` with double quotes.

### Expected result

The command line arguments take precedence over the configuration and Rome formats the string literals in `jquery.min.js` with single quotes 

### Code of Conduct

- [X] I agree to follow Rome's Code of Conduct

### PR DESCRIPTION
**Pull request by ematipico**
## Summary

Fixes #3175 

We were incorrectly computing a `WorkspaceSettings`, while now we should change/create a `Configuration` struct instead and then update the settings of the workspace.

This bug went undetected because we don't have tests that check that arguments passed via CLI are correctly applied.

## Test Plan

I created two new tests, one that applies the generic configuration of the formatter and one that applies the configuration for the javascript language.

### CODE REVIEW

File: crates/rome_cli/src/commands/check.rs [hunk 1]
```diff
@@ -31,13 +28,11 @@ pub(crate) fn check(mut session: CliSession) -> Result<(), Termination> {
...
 
-    if let Some(configuration) = configuration {
-        session
-            .app
-            .workspace
-            .update_settings(UpdateSettingsParams { configuration })?;
-    }
-    apply_format_settings_from_cli(&mut session, &mut workspace_settings)?;
+    let configuration = apply_format_settings_from_cli(&mut session, configuration)?;
+    session
```
Comment (MichaReiser) [inline crates/rome_cli/src/commands/check.rs:32]: What does this change mean for concurrent usages of the deamon. Do the CLI argument change for all requests to the workspace?

Comment (ematipico) [reply crates/rome_cli/src/commands/check.rs:32]: Yes

**Review by leops — APPROVED** [review]
This looks good overall, can we just add an additional test that has both a configuration file and CLI arguments override to ensure the priority is handled correctly ?

File: crates/rome_cli/src/commands/check.rs (hunk 1 above)
Comment (MichaReiser) [reply crates/rome_cli/src/commands/check.rs:32]: Yes, but what if multiple clients are connected to the same daemon 

```bash
# client 1
rome format --indent-style="tab" --use-server

#client 2
rome format --indent-style="space" --use-server
```


And both commands run at the same time. Which configuration would be used?

Comment (ematipico) [reply crates/rome_cli/src/commands/check.rs:32]: Both clients have two different instances of `Workspace`, so their configuration won't mix up.

**Pull request merged**